modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -196,13 +196,14 @@ private synchronized void createSource(boolean restarting) throws Exception {
     }
   }
 
-  private void startSource(boolean restarting) throws MuleException {
+  private void startSource(boolean restarting, Map<String, Object> restartingContext) throws MuleException {
```

Comment:
don't use maps. Build a explicit pojo.

Reply:
In the end I would need to use a POJO with at least a field that is a map, right? What is stored in the context is specific to the restarting of a polling source.

Reply:
why? right now you're using a map on which you only put two entries with well known keys. Just use a pojo with two well known fields... If in the future you need a map, you add a map, but I don't see why you would need that and you certainly don't need it now

Reply:
My point is, the Restartable interface needs to be generic, not everything that can be restarted needs a Runnable and a Scheduler. Does it make sense?

Reply:
but this is internal. We can change it as much as we want.. Rename it to `RestartableSource` if you want